tests/phpunit/Authentication/OAuth10Test.php

```diff
@@ -6,55 +6,69 @@
 
 use Keboola\GenericExtractor\Authentication\OAuth10;
 use Keboola\Juicer\Client\RestClient;
+use Keboola\Juicer\Tests\HistoryContainer;
+use Keboola\Juicer\Tests\RestClientMockBuilder;
 use PHPUnit\Framework\TestCase;
-use Psr\Log\NullLogger;
 
 class OAuth10Test extends TestCase
 {
-    protected function setUp(): void
-    {
-        $this->markTestSkipped('TODO fix test');
-        parent::setUp();
-    }
-
     public function testAuthenticateClient(): void
     {
-        $restClient = new RestClient(new NullLogger(), [], [], []);
-        $auth = new OAuth10(
-            [
+        // Create Oauth10 auth
+        $authorization = [
             'oauth_api' => [
                 'credentials' => [
                     '#data' => '{"oauth_token": "token", "oauth_token_secret": "token_secret"}',
                     'appKey' => 'aaa',
                     '#appSecret' => 'bbb',
                 ],
             ],
-            ]
-        );
-        $auth->attachToClient($restClient);
+        ];
+        $auth = new OAuth10($authorization);
 
-        self::assertEquals('oauth', $restClient->getClient()->getDefaultOption('auth'));
+        // Create RestClient
+        $history = new HistoryContainer();
+        $restClient = RestClientMockBuilder::create()
+            ->addResponse200((string) json_encode((object) [
+                'data' => [1, 2, 3],
+            ]))
+            ->addResponse200((string) json_encode((object) [
+                'data' => [4, 5, 6],
+            ]))
+            ->setHistoryContainer($history)
+            ->setInitCallback(function (RestClient $restClient) use ($auth): void {
+                $auth->attachToClient($restClient);
+            })
+            ->getRestClient();
 
-        $request = $restClient->createRequest(['endpoint' => '/']);
-
-        $mock = new Mock(
-            [
-            new Response(200, [], Stream::factory('{}')),
-            ]
+        // Each request contains Authorization header
```

Comment:
Samozny test OAuth10Test je iba prepisany do novej formy, ... a pre istotu testujem 2 requesty, teda ze kazdy obsahuje hlavicku -> tak je to aj v inych testoch.